src/task/task.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { Repository } from "typeorm";
+
+import { TaskDTO } from "./task.dto";
+import { Task } from "./task.entity";
+
+@Injectable()
+export class TaskService {
+
+	public constructor(
+		@InjectRepository(Task)
+		private taskRepo: Repository<Task>
+	) { }
+
+	public insertTask(taskDTO: TaskDTO): void {
+		this.taskRepo.insert(taskDTO);
+	}
+
+	public async getAllTasks(): Promise<Task[]> {
+		return this.taskRepo.find();
+	}
+
+	public async getTaskById(id: number): Promise<Task> {
+		return this.taskRepo.findOne({ where: { id: id } });
+	}
+
+	public async updateTaskById(taskDTO: TaskDTO): Promise<TaskDTO> {
+		await this.taskRepo.update(taskDTO.id, taskDTO);
+		return this.taskRepo.findOne({ where: { id: taskDTO.id } });
+	}
```

Comment:
findOne will return a Task Entity. You should return a TaskDTO.
To do this, you should add a new service named TaskMapper. TaskMapper will have several methods:
toDTO(entity: Task): TaskDTO{}, toEntity(dto: TaskDTO): Task {}

Reply:
please update everywhere